docs/guide/pagination.md

```diff
@@ -12,20 +12,190 @@ An interface for limit/offset pagination can be use for basic pagination needs:
 query {
   fruits(pagination: { offset: 0, limit: 2 }) {
     name
-    color
   }
 }
 ```
 
-There is not default limit defined. All elements are returned if no pagination limit is defined.
+The `pagination` argument can be given to the type, which will enforce the pagination
+argument every time the field is annotated as a list, but you can also give it directly
+to the field for more control, like:
+
+```python title="types.py"
+@strawberry_django.type(models.Fruit)
+class Fruit:
+    name: auto
+
+
+@strawberry.type
+class Query:
+    fruits: list[Fruit] = strawberry_django.field(pagination=True)
+```
+
+Which will produce the exact same schema.
+
+> [!NOTE]
+> There is no default limit defined. All elements are returned if no pagination limit is defined.
+
+## Paginated Generic
+
+For more complex pagination needs, you can use the `Paginated` generic, which alongside
+the `pagination` argument, will wrap the results in an object that contains the results
+and the pagination information, together with the `totalCount` of elements excluding pagination.
+
+```python title="types.py"
+from strawberry_django.pagination import Paginated
+
+
+@strawberry_django.type(models.Fruit)
+class Fruit:
+    name: auto
+
+
+@strawberry.type
+class Query:
+    fruits: Paginated[Fruit] = strawberry_django.field()
+```
+
+Would produce the following schema:
+
+```graphql title="schema.graphql"
+type Fruit {
+  name: String!
+}
+
+type PaginatedInfo {
+  limit: Int!
+  offset: Int!
+}
+
+type FruitPaginated {
+  pageInfo: PaginatedInfo!
+  totalCount: Int!
+  results: [Fruit]!
+}
+
+type Query {
+  fruits(pagination: PaginationInput): [FruitPaginated!]!
+}
+```
+
+Which can be queried like:
+
+```graphql title="schema.graphql"
+query {
+  fruits(pagination: { offset: 0, limit: 2 }) {
+    totalCount
+    pageInfo {
+      limit
+      offset
+    }
+    results {
+      name
+    }
+  }
+}
+```
+
+### Customizing the pagination
+
+Like other generics, `Paginated` can be customized to modify its behavior or to
+add extra functionality in it. For example, suppose we want to add the average
+price of the fruits in the pagination:
+
+```python title="types.py"
+from strawberry_django.pagination import Paginated
+
+
+@strawberry_django.type(models.Fruit)
+class Fruit:
+    name: auto
+    price: auto
+
+
+@strawberry.type
+class FruitPaginated(Paginated[Fruit]):
+    @strawberry_django.field
+    def average_price(self) -> Decimal:
+        if self.queryset is None:
+            return Decimal(0)
+
+        return self.queryset.aggregate(Avg("price"))["price__avg"]
+
+    @strawberry_django.field
+    def paginated_average_price(self) -> Decimal:
+        paginated_queryset = self.get_paginated_queryset()
+        if paginated_queryset is None:
+            return Decimal(0)
+
+        return paginated_queryset.aggregate(Avg("price"))["price__avg"]
+
+
+@strawberry.type
+class Query:
+    fruits: FruitPaginated = strawberry_django.field()
```

Comment:
`strawberry_django.field()` is mandatory here, right?

Reply:
Yes it is, otherwise `@strawberry.type` wouldn't know that this needs to be set to a `StrawberryDjangoField` and not a `StrawberryField`.

something that we can maybe try to improve in the future?